// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";

package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

message CloudbillingProjectBillingInfo {
  string name = 1;
  string billing_account_name = 2;
  string billing_enabled = 3;
}

message ApplyCloudbillingProjectBillingInfoRequest {
  CloudbillingProjectBillingInfo resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteCloudbillingProjectBillingInfoRequest {
  string service_account_file = 1;
  CloudbillingProjectBillingInfo resource = 2;
}

message ListCloudbillingProjectBillingInfoRequest {
  string service_account_file = 1;
}

message ListCloudbillingProjectBillingInfoResponse {
  repeated CloudbillingProjectBillingInfo items = 1;
}

service CloudbillingProjectBillingInfoService {
  rpc ApplyCloudbillingProjectBillingInfo(ApplyCloudbillingProjectBillingInfoRequest) returns (CloudbillingProjectBillingInfo);
  rpc DeleteCloudbillingProjectBillingInfo(DeleteCloudbillingProjectBillingInfoRequest) returns (google.protobuf.Empty);
  rpc ListCloudbillingProjectBillingInfo(ListCloudbillingProjectBillingInfoRequest) returns (ListCloudbillingProjectBillingInfoResponse);
}
